client-java/controller/src/test/java/org/evomaster/client/java/controller/internal/db/mongo/MongoTestTemplate.java

```diff
@@ -0,0 +1,83 @@
+package org.evomaster.client.java.controller.internal.db.mongo;
+
+import com.mongodb.MongoClient;
+import io.restassured.http.ContentType;
+import org.evomaster.client.java.controller.InstrumentedSutStarter;
+import org.evomaster.client.java.controller.api.dto.SutRunDto;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.testcontainers.containers.GenericContainer;
+
+import static io.restassured.RestAssured.given;
+import static org.evomaster.client.java.controller.api.ControllerConstants.*;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public abstract class MongoTestTemplate {
+
+    protected static final int MONGO_PORT = 27017;
+    protected static GenericContainer mongo;
+    protected static MongoClient mongoClient;
+
+    @BeforeAll
+    public static void startMongoContainer() {
+        mongo = new GenericContainer("mongo:3.2")
+                .withExposedPorts(MONGO_PORT);
+        mongo.start();
+        mongoClient = new MongoClient(mongo.getContainerIpAddress(), mongo.getMappedPort(MONGO_PORT));
+    }
+
+    @AfterAll
+    public static void stopMongoContainer() {
+        if (mongo != null) {
+            mongo.stop();
+        }
+    }
+
```

Comment:
one thing we will need to find out is to how to "clean/reset" the state of Mongo (something like we do for SQL in `DbCleaner`). 
once we know it, then we ll need to call it here in a `BeforeEach`, or, if testing in the controller, have it in its `reset` method

Reply:
ok. I will drop all databases in this super class